assign_rights/assemble.py

```diff
@@ -72,10 +55,16 @@ def calculate_dates(self, object, request_start_date, request_end_date):
                 representing the group of objects' start and end dates after
                 calculation.
         """
-        object_start = self.get_date_value(
-            object, "start_date", request_start_date, object.start_date_period)
-        object_end = None if object.end_date_open else self.get_date_value(
-            object, "end_date", request_end_date, object.end_date_period)
+        object_start = None
+        object_end = None
+        if getattr(object, "start_date_period"):
```

Comment:
I would bet this is the offending line. Probably needs to be 
```suggestion
        if getattr(object, "start_date_period") is not None:
```

Reply:
I fixed this another way before seeing this--I flipped the order of which type of start is evaluated first.

Reply:
OK, that's another way of solving that problem!